pagarme/react-style-guide
# Add new validations to code standards

This PR adds a few new validations over `react hooks` and styles and upgrades some dependencies. Also, improves the `read.me`

## How to test

1. Clone this project
```
git clone -b update/dependencies git@github.com:pagarme/react-style-guide.git
```

2. Go to the eslint project folder and create a yarn link(this should create a link called `eslint-config-pagarme-react`)
```
  cd react-style-guide/eslint
  yarn link
```

3. Do de same for the stylelint project(this should create a link called `stylelint-config-pagarme-react`)
```
  cd react-style-guide/stylelint
  yarn link
```

4. In [Former kit project](https://github.com/pagarme/former-kit) link the created refences of `eslint-config-pagarme-react` and `stylelint-config-pagarme-react` using the following commands:
```
  yarn link eslint-config-pagarme-react
  yarn link stylelint-config-pagarme-react
```

5. In [Former kit folder](https://github.com/pagarme/former-kit) install all dependencies including the new dependencies related to the added packs.
```
  yarn
```

6.  Run the lint command to test the new rules. This should trigger some errors related to the new rules.
```
  yarn lint
```

This process should create some errors and ask for some packages installation(probably `eslint-plugin-react-hooks`). The final result using the [Former kit's](https://github.com/pagarme/former-kit) master branch should be similar to the following picture:
![Screen Shot 2019-04-12 at 19 15 20](https://user-images.githubusercontent.com/1773766/56069009-646e5100-5d57-11e9-8500-d509196d4674.png)
A lot of lint errors 😄
